packages/ui/src/components/multi-select-v2.tsx

```diff
@@ -0,0 +1,329 @@
+            }
+          }
+          if (e.key === 'Enter' && input.value && !disallowCreation) {
+            if (
+              !options.some(option => option.key.toLowerCase() === input.value.toLowerCase()) &&
+              !(isControlled ? value : selected).some(s => s.key.toLowerCase() === input.value.toLowerCase())
+            ) {
+              let newOption: MultiSelectOption
+
+              if (input.value.includes(':')) {
+                const [key, value] = input.value.split(':', 2)
+
+                if (key && key.trim()) {
+                  newOption = {
+                    key: key.trim(),
+                    value: value ? value.trim() : '',
+                    id: input.value
+                  }
+                } else {
+                  newOption = {
+                    key: input.value,
+                    id: input.value
+                  }
+                }
+              } else {
+                newOption = {
+                  key: input.value,
+                  id: input.value
+                }
+              }
+              const newOptions = [...(isControlled ? value : selected), newOption]
+              if (isControlled) {
+                onChange?.(newOptions)
+              } else {
+                setSelected(newOptions)
+              }
+              setInputValue('')
+              setSearchQuery?.('')
+              e.preventDefault()
+            }
+          }
+          if (e.key === 'Escape') {
+            input.blur()
+          }
+        }
+      },
+      [handleUnselect, selected, disallowCreation, onChange, options, setInputValue, isControlled, value]
+    )
+
+    useEffect(() => {
+      if (open) {
+        document.addEventListener('mousedown', handleClickOutside)
+        document.addEventListener('touchend', handleClickOutside)
+      } else {
+        document.removeEventListener('mousedown', handleClickOutside)
+        document.removeEventListener('touchend', handleClickOutside)
+      }
+
+      return () => {
+        document.removeEventListener('mousedown', handleClickOutside)
+        document.removeEventListener('touchend', handleClickOutside)
+      }
+    }, [open])
+
+    useEffect(() => {
+      setOptions(arrayOptions ?? [])
+    }, [arrayOptions])
+
+    const [availableOptions, setAvailableOptions] = useState<MultiSelectOption[]>([])
+
+    useEffect(() => {
+      if (!options || options.length === 0) {
+        setAvailableOptions([])
+        return
+      }
+
+      const filteredOptions = options.filter(option =>
+        isControlled
+          ? !value?.some(selectedOption => selectedOption.id === option.id)
+          : !selected?.some(selectedOption => selectedOption.id === option.id)
+      )
+
+      setAvailableOptions(filteredOptions)
+    }, [options, selected, isControlled, value, inputValue, searchQuery, open])
+    return (
+      <div className="cn-multi-select-outer-container">
+        <Label disabled={disabled}>{label}</Label>
```

Comment:
maybe form elements like Label, Caption, error, warning etc can be moved to the FormWrapper